inst/htmljs/animint.js

```diff
@@ -2019,13 +2030,91 @@ var animint = function (to_select, json_file) {
       d3.select("title").text(response.title);
     }
     // Add plots.
+  
+  
+  var check_svg = function (row_number,col_number) {
+    var id="#row"+row_number+"col"+col_number
+    var parentElement= d3.select(id)
+    if (!parentElement.select('svg').empty()) {
+      return true
+    }else{
+      return false
+    }
+  }
+  
+  
+  var maximum_row= 0
+  var maximum_column= 0
+  var construct_outer_table = function () {
+  var count=0
+  
+  var find_maximum_dimensions = function () {
+  
+  
     for (var p_name in response.plots) {
-      add_plot(p_name, response.plots[p_name]);
+      count=count+1
+    if (response.plots[p_name].row>maximum_row){
+      maximum_row=response.plots[p_name].row
+    }
+    if (response.plots[p_name].col>maximum_column){
+      maximum_column=response.plots[p_name].col
+    }
+  }
+   }
+  find_maximum_dimensions()
+    var count_dimensions = Math.ceil(Math.sqrt(count))-1;
+    var outer_table = element.append("table").attr("id", "outerTable").style("width", "1500px").style("border", "1px solid white");
+    maximum_row=Math.max(maximum_row,count_dimensions)
+    maximum_column=Math.max(maximum_column,count_dimensions)
+    //console.log(maximum_row)
+    //console.log(maximum_column)
+    for (var i = 0; i <=maximum_row; i++) {
+        var current_row = outer_table.append("tr").style("border", "1px solid white").style("height", "500px");
+        for (var j = 0; j <=maximum_column; j++){
+            current_row.append("td").attr("id", "row"+i+"col"+j).style("border", "1px solid white").style("width", "500px");
+        }
```

Comment:
or probably better to just leave them undefined, and the height/width will stretch to whatever size is needed

Reply:
the cell in the outer table already stretches to more size if plot has larger dimension than height width. but I will still double check that. I hardcoded 500 for width and height because the default hardcoded dimensions of plots are are 400 by 400. so using 500 by 500 can be ideal.

Reply:
i would rather avoid adding more arbitrary constants like 500 unless absolutely necessary (please specify/explain if that is the case)

Reply:
the highlighted lines of code create outer html table. during creation of the table since we do not know the dimensions of plot that will be going in a specific cell, I create a table with default cell dimensions of 500px. 500px because I think it is better to give some extra space for default dimensions of 400px for each plot if user has not passed width and height. if plot is bigger then D3 Javascript handles that by overriding the 500px value. what values would you recommend for each cell while creating the outer table ?

Reply:
please "leave them undefined, and the height/width will stretch to whatever size is needed"

if the justification is "I think it is better to give some extra space " then we should create some way to customize that space, but that seems overly complicated for a first try at this functionality.

I think we should leave it undefined for now, and introduce more complex code for configuring that later, if we find it to be really necessary.

another issue with 500px for each cell is that we may want to have plots which are smaller than that, which would leave lots of undesirable empty space.

Reply:
Okay. I would do that

Reply:
> please "leave them undefined, and the height/width will stretch to whatever size is needed"
> 
> if the justification is "I think it is better to give some extra space " then we should create some way to customize that space, but that seems overly complicated for a first try at this functionality.
> 
> I think we should leave it undefined for now, and introduce more complex code for configuring that later, if we find it to be really necessary.
> 
> another issue with 500px for each cell is that we may want to have plots which are smaller than that, which would leave lots of undesirable empty space.

this was done. cells are now no longer with hardcoded values